### PR TITLE
Handle VM disconnection gracefully and reorder termination logic

### PR DESCRIPTION
**Pull request by zepedrocosta**
After the changes in #620 the stop button wasn't working properly. The action wasn't executed and the process just kept running.

The `terminate()` now is called before `detach()` when terminating the debuggee, ensuring proper shutdown sequence.

### CODE REVIEW

**Comment by zepedrocosta** [comment]
> @zepedrocosta please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

@microsoft-github-policy-service agree

File: com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/DisconnectRequestHandler.java [hunk 1]
```diff
@@ -26,8 +26,8 @@ public void destroyDebugSession(Command command, Arguments arguments, Response r
         IDebugSession debugSession = context.getDebugSession();
         if (debugSession != null) {
             if (disconnectArguments.terminateDebuggee && !context.isAttached()) {
-                debugSession.detach();
                 debugSession.terminate();
+                debugSession.detach();
```
Comment (wenytang-ms) [inline com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/DisconnectRequestHandler.java]: In case of an exception, detach() will not be executed, it's better to 
```java
try {
       debugSession.terminate();
} finally {
       debugSession.detach();
}
```

**Review by wenytang-ms — APPROVED** [review]

**Comment by wenytang-ms** [comment]
@zepedrocosta thank you for your contribute!
